### PR TITLE
feat: feat: Edit report name - Added mixpanel trackers + Success toast message with code coverage for Approver persona

### PR DESCRIPTION
**Pull request by Dimple16**
### <samp>🤖 Generated by Copilot at ad4519a</samp>

This pull request adds and updates mock data, enums, models, services, components, and unit tests for various features of the fyle-mobile-app, such as unflattened expenses, per diem, mileage, report name change, and employee details. It also fixes some TypeScript errors, code formatting, and readability issues.

### <samp>🤖 Generated by Copilot at ad4519a</samp>

> _We mock the data, we import the enums_
> _We test the features, we fix the errors_
> _We track the events, we change the names_
> _We code with fury, we fyle with flames_

### <samp>🤖 Generated by Copilot at ad4519a</samp>

*  Add mock data arrays and objects for testing mileage, per diem, and expense categories and properties ([link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-8914bfb32bd70efe52e673bc9938855b2c84a494d9464f82c50fe2c8bfbdca45R1002-R1040), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-9f195d091788e9bb0c8df097ef4fcffbee53b84e408eda5858d11b718eb2d7c4R1-R102), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-db737e50777751787c68eda6c5be391ed672eb00607a92de6aaf9c945a3010d6R563-R584), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-21ae928ce640ad62cfada5fee7bf0bbc490300ab04a25f8309a9878e21b2f1cfR2750-R2759), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15L633-R633), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15L682-R682), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15L697-R697), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15L752-R752), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15L768-R768), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57R17-R20), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-5ea8feba7a8355dcc3a878d59e6ac3c56d89e15472f7ddfe5d9c63c38a90c961L9-R20))
* Add enums of expense source and state values to the models folder and import them in the mock data file ([link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-db737e50777751787c68eda6c5be391ed672eb00607a92de6aaf9c945a3010d6R1-R2), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-ab3c3d797d98910ab60186846df3ccc8d8ad4ea0c12b7c31cd58bb26e569aec0R1-R11), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-a910c06bac380de1af218aaf136f702c3f295f9016f66ad9637d934e465dfe97R1-R9))
* Add interface of report name change properties to the models folder and import it in the tracking service file ([link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-0c72ffb386f278313e6e26aaa670c598d8c1ce5437fb03d62da7a09cabbea44eR142-R146), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-63225a1ea9f44d74739fc2787c222f8a94585abcf21a01bf159adbb4c6597d7aR27))
* Add method of reportNameChange to the tracking service and call it with the event name and properties object ([link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-63225a1ea9f44d74739fc2787c222f8a94585abcf21a01bf159adbb4c6597d7aR226-R230))
* Modify the type of attachments$ observable to avoid TypeScript error and add a condition to handle empty comment for policy violation in the add-edit-expense page ([link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL268-R268), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bR3926-R3928))
* Add type assertions and optional chaining operators to avoid TypeScript errors and possible errors of accessing undefined properties in the add-edit-mileage and employee-details-card components ([link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL784-R784), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L284-R284), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-14bb43ace42b08aac6b74497dbe5dd4868c53d31110312c6c40b11caaa690684L17-R17))
* Add test cases for the methods of goToTransaction, getProjectCategoryIds, and saveExpenseAndGotoNext in the add-edit-mileage-1.spec.ts file ([link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-5ea8feba7a8355dcc3a878d59e6ac3c56d89e15472f7ddfe5d9c63c38a90c961R299-R477))
* Split the test cases for the add-edit-per-diem page into smaller files and execute them in the add-edit-per-diem.page.setup.spec.ts file ([link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-e07d5851dc6ce6b9d4929a5a7cd5cce88c19fb320d9efbd769f8f1911bbd440dR1-R276), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-113c190c8d5e8c3f8c98cf1e699be2562159cd2ba8d04e2e19a31ef5b74d35c2R39), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-113c190c8d5e8c3f8c98cf1e699be2562159cd2ba8d04e2e19a31ef5b74d35c2R283))
* Remove unused imports, spies, and empty lines of code to avoid linting errors and improve code readability and formatting ([link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L103-R103), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L1157), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L1175), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L1187), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L1205), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L1215-R1207), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L77-L80), [link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-ba304f41b9ceb84ec3edd25c39133e53d238dc01ad0cb7b04fbd999aa2300639L1345))
* Add return types of void to the methods of onUpdateMobileNumber and onVerifyMobileNumber in the employee-details-card component ([link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-99d36498885afe696ab2bff61cc05cfc9fd8f025c757e673e86ebe9a29ca12eeL16-R23))
* Add a type assertion of never to the perDiem.id property in the per-diem service file ([link](https://github.com/fylein/fyle-mobile-app/pull/2278/files?diff=unified&w=0#diff-65d87a758a019ac2bfac64e03b32b409b89f9a045b70b0dc38f69c32749f52e7L52-R52))

## Clickup
https://app.clickup.com/t/85zttv251

## Code Coverage
Please add code coverage here

## UI Preview
Same changes as Spender - https://github.com/fylein/fyle-mobile-app/pull/2280

### CODE REVIEW

**Review by abhishek1234321 — CHANGES_REQUESTED** [review]

File: src/app/fyle/view-team-report/view-team-report.page.ts [hunk 1]
```diff
@@ -576,6 +586,24 @@ export class ViewTeamReportPage implements OnInit {
     }
   }
 
+  trackReportNameChange(): void {
```
[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/fyle/view-team-report/view-team-report.page.ts:589]: yes, this method is in 2 pages, but the persona is different, spender and approver,
the method is very specific to that page as we are tracking start and end time, etc,
adding this page-specific method to a service will generalize it no? imo I don’t think we should move the trackReportNameChange to a service
for the successToast method, if we move it to a service, we’ll again have to import matSnackBar,

[the remark this answers is not in the document]
Comment (abhishek1234321) [reply src/app/fyle/view-team-report/view-team-report.page.ts:589]: right, but this is mobile app, we don't have a concept of different persona here right?

[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/fyle/view-team-report/view-team-report.page.ts:589]: yes, but, why add details like tracking end time to a service? we'll have to define variables related to this page in the report service,
is that okay? we'll have to pass 4 variables

[the remark this answers is not in the document]
Comment (abhishek1234321) [reply src/app/fyle/view-team-report/view-team-report.page.ts:589]: hmm right, theek hai approving

**Review by abhishek1234321 — APPROVED** [review]

**Pull request merged**